fontforge/python.c

```diff
@@ -1735,50 +1735,100 @@ return( Py_BuildValue("i",ret));
-	PyMem_Free(title);
-	PyMem_Free(quest);
-return( NULL );
+        PyErr_Format(PyExc_TypeError, "Expected a tuple of strings for the third argument");
+        PyMem_Free(title);
+        PyMem_Free(quest);
+        return( NULL );
     }
+
     cnt = PySequence_Size(answero);
-    answers = calloc(cnt+1, sizeof(char *));
-    if ( def<0 || def>=cnt ) {
-	PyErr_Format(PyExc_ValueError, "Value out of bounds for 4th argument");
-	PyMem_Free(title);
-	PyMem_Free(quest);
-return( NULL );
+    char* sel = calloc(cnt, sizeof(char));
+
+    if (defo != NULL && defo != Py_None)
+    if (multiple) {
+        if ( !PyTuple_Check(defo) ) {
+            PyErr_Format(PyExc_TypeError, "4th argument must be a tuple" );
+            PyMem_Free(title); PyMem_Free(quest); free(sel);
+            return( NULL );
+        }
+        int dcnt = PySequence_Size(defo);
+        if (dcnt != cnt) {
+            PyErr_Format(PyExc_ValueError, "Expected tuple/list of %d items, got %d items", cnt, dcnt );
+            PyMem_Free(title); PyMem_Free(quest); free(sel);
+            return( NULL );
+        }
+        for (int i = 0; i < cnt; i++) {
+            PyObject* temp = PyTuple_GetItem(defo, i);
+            sel[i] = temp == Py_True ? (char)1 : (char)0;
+        }
+    } else {
+        if (!PyInt_Check(defo) && !PyLong_Check(defo)) {
+            PyErr_Format(PyExc_TypeError, "Expected an integer for 4th argument");
+            PyMem_Free(title); PyMem_Free(quest); free(sel);
+            return( NULL );
+        }
+        def = (int)PyLong_AsLong(defo);
+        if ( def<0 || def>=cnt ) {
+            PyErr_Format(PyExc_ValueError, "Value out of bounds for 4th argument");
+            PyMem_Free(title); PyMem_Free(quest); free(sel);
+            return( NULL );
+        }
     }
-    for ( i=0; i<cnt; ++i ) {
+
+    answers = calloc(cnt+1, sizeof(char *));
+    for ( int i=0; i<cnt; ++i ) {
         PyObject *utf8_name = PYBYTES_UTF8(PySequence_GetItem(answero,i));
         if ( utf8_name==NULL ) {
-	    PyMem_Free(title);
-	    PyMem_Free(quest);
-	    FreeStringArray( i, answers );
-return( NULL );
-	}
+            PyMem_Free(title); PyMem_Free(quest); free(sel);
+            FreeStringArray( i, answers );
+            return( NULL );
+        }
         answers[i] = copy(PyBytes_AsString(utf8_name));
         Py_DECREF(utf8_name);
     }
     answers[cnt] = NULL;
 
-    ret = ff_choose(title,(const char **) answers,cnt,def,quest);
-    PyMem_Free(title);
-    PyMem_Free(quest);
+    PyObject* reto;
+    if (multiple) {
+        char* buts2[2] = {_("OK"), _("Cancel")};
+        reto = PyTuple_New(cnt);
+        ret = ff_choose_multiple(title,(const char **) answers,sel,cnt,buts2,quest);
+        for (int i = 0; i < cnt; i++) {
+            char o = sel[i];
+            PyTuple_SetItem(reto, i, o == 1 ? Py_True : Py_False);
+            Py_INCREF(o == 1 ? Py_True : Py_False); // prevents crash on finalize
```

Comment:
Weird that this is needed but the docs bear it out. 

I think here I would do `PyObject o = (sel[i] == 1) ? PyTrue : Py_False;` and then just use `o` in the following two lines.